### ISSUE TITLE
Unable to pass ref to Datepicker [react-aria-components]

### ISSUE THREAD

**Issue by wmodden** [adobe/react-spectrum#7756]
### Provide a general summary of the issue here

I have a custom [Datepicker](https://react-spectrum.adobe.com/react-aria/DatePicker.html) component that I'm integrating with `react-hook-form`. I was wondering which component should received the ref (passed by forwardRef), I need to do so since `react hook form` uses this ref to focus an element on error (think a form submission that has client validation)

I've tried with the `DatePicker` and `DateInput` but no luck.

### 🤔 Expected Behavior?

Ref should be received by the `DatePicker` or `DateInput` component.

### 😯 Current Behavior

Looks like the ref is not being received by the mentioned react-aria-components.

### 🖥️ Steps to Reproduce

You can use the documentation https://react-spectrum.adobe.com/react-aria/DatePicker.html

### Version

1.5.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macos sequoia

**Comment by reidbarber** [comment]
Looks like it is being added here:

https://github.com/adobe/react-spectrum/blob/4b8b33a022a17c7df2d6ee6d23b6208b503973fe/packages/react-aria-components/src/DatePicker.tsx#L162-L166

Are you seeing different behavior? You can try to `console.log` the ref to check.

**Comment by adhameer** [comment]
I am experiencing the same issue with `DateField`. I need to focus the actual "input" (which is actually the first of multiple inputs in a group) and it seems to require quite a bit of work to attach the ref to the right place regardless of the user's locale.

The refs passed directly to `DateField` and `DateInput` are applied to the outer elements, which aren't the focusable ones. What _does_ work is passing the ref to the first `DateSegment`. However, there are no values in the FaC of `DateInput` that allow me to figure out which segment is the first one. An easy way to do it is as follows:

```tsx
function MyCustomDateField() {
  const ref = React.useRef(null);

  return (
    <DateField>
      <Label>Label</Label>
      <DateInput>
        {(segment) => <DateSegment segment={segment} ref={segment.type === 'month' ? ref : undefined} />}
      </DateInput>
    </DateField>
  );
}
```

but of course this is making an assumption about the user's locale and is inflexible. Probably I can use the internationalization library to determine which is the type of the first segment, but that would be a lot of work for this simple functionality.

Something that would help would be if the FaC included a value that indicated if the segment being rendered is the first editable segment of the input. A simple index passed in could work, though I'm not sure if there are any cases where the first segment is actually a literal (which cannot be focused - we need the first _editable_ segment).

For instance:

```tsx
<DateInput>
  {(segment, index) => <DateSegment segment={segment} ref={index === 0 ? ref : undefined} />}
</DateInput>
```

But ideally there would just be one simple place to attach the ref as it is for other non-compound form inputs.

**Comment by joestrkr** [comment]
I have the exact same issue, trying to pass the ref from react-hook-form. 

Weird thing is that tabbing into the DateField with the keyboard or clicking on the label or DateField group element correctly places the focus on the first DateSegment. It just does not seem to work when RHF focuses the element programmatically.

**Comment by adhameer** [comment]
@wmodden @joestrkr I've found a solution that doesn't require _too_ much extra code and seems to work for me for now. Basically, we make use of the `DateFieldStateContext` to find out which segment is the first editable segment, and then we can use that information to pass the ref to only that segment.

The code looks something like this:

```ts
const MyCustomDateInput = React.forwardRef<HTMLInputElement>(function DateInput(_props, ref) {
  const { segments } = useContext(DateFieldStateContext)!;
  // Find the first editable date segment - this is the one we want to focus
  // when focusing the input
  const firstSegmentType = segments.find(
  	({ isEditable }) => isEditable,
  )?.type;

  return (
    <DateInput>
      {(segment) => (
        <DateSegment segment={segment} ref={segment.type === firstSegmentType ? ref : undefined} />
      )}
    </DateInput>
  );
});

function MyCustomDateField() {
  const ref = React.useRef<HTMLInputElement>(null);

  return (
    <DateField>
      <Label>Label</Label>
      <MyCustomDateInput ref={ref} />
    </DateField>
  );
}
```

`MyCustomDateField` would then be updated to use a forwarded ref.

**Comment by wmodden** [comment]
I still see this on version 1.11.0

**Comment by wmodden** [comment]
> Maybe we could add a `focusableRef` prop or something like that to DateInput? We're pretty consistent across all of our components that the forwarded `ref` always goes to the outer-most element. But I could see for this specific case of focusing the first segment it's not super easy at the moment.

happy to help if you can show me the way (some of it of course) 🚀

**Comment by lensbart** [comment]
> Maybe we could add a `focusableRef` prop or something like that to DateInput? We're pretty consistent across all of our components that the forwarded `ref` always goes to the outer-most element. But I could see for this specific case of focusing the first segment it's not super easy at the moment.

Another benefit of having a `ref` that points to the input element, is that it allows using the input value via `ref.current?.value` without requiring a controlled component.

**Comment by hasegawa-101** [comment]
Hi! @snowystinger I'd like to work on this issue. I'll submit a PR soon.

**Comment by LaurAndreea10** [comment]
Hi! I’d like to take this one.
I’ll start by reproducing the issue with a small react-hook-form example, then I’ll check which component should correctly receive the forwarded ref and propose a focused fix with tests if possible.

**Comment by rigobauer** [comment]
@devongovett Any plans to fix this? Maybe there's a reason we're not seeing, but looks like the PR was ready? Thanks
